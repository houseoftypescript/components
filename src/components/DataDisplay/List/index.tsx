import styled from 'styled-components';

export const List = styled.div`
  border: 1px solid #d1d5db;
  border-radius: 0.25rem;
  overflow: hidden;
`;

export type ListItemProps = {
  last?: boolean;
};

export const ListItem = styled.div<ListItemProps>`
  ${(props: ListItemProps) =>
    props.last ? '' : 'border-bottom: 1px solid #d1d5db;'}
  padding: 1rem;
`;

export default Object.assign(List, {
  Item: ListItem,
});
