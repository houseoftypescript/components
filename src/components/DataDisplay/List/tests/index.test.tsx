import { render } from '@testing-library/react';
import React from 'react';
import List from '../index';

describe('List', () => {
  it('to match snapshot', () => {
    const { container } = render(
      <List>
        <List.Item>Item</List.Item>
        <List.Item>Item</List.Item>
        <List.Item last>Item</List.Item>
      </List>
    );
    expect(container).toMatchSnapshot();
  });
});
