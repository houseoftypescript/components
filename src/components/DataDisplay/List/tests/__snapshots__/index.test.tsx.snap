// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List to match snapshot 1`] = `
<div>
  <div
    class="sc-beySbM eigaaq"
  >
    <div
      class="sc-guDLRT jlkDGc"
    >
      Item
    </div>
    <div
      class="sc-guDLRT jlkDGc"
    >
      Item
    </div>
    <div
      class="sc-guDLRT bPHRUW"
    >
      Item
    </div>
  </div>
</div>
`;
